src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -217,6 +218,7 @@ def _get_azure_cli_properties(self):
         set_custom_properties(result, 'RegionInput', self.region_input)
         set_custom_properties(result, 'RegionIdentified', self.region_identified)
         set_custom_properties(result, 'AllowBroker', str(self.allow_broker))
+        set_custom_properties(result, 'MsalTelemetry', str(self.msal_telemetry))
```

Comment:
Is `self.msal_telemetry` string type or other object type? Wondering why need `str()` here

Reply:
Telemetry returned by MSAL Python is a string. It is also OK for Azure CLI to `str(...)`, so that it is more robust.

Reply:
Other properties don't have the `str()` casting. Even though `str()` makes it more robust, it may hide errors if MSAL no longer returns a `str`, so let's remove it for consistency:

```suggestion
        set_custom_properties(result, 'MsalTelemetry', self.msal_telemetry)
```